src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -938,13 +956,6 @@ ExecuteSingleSelectTask(CitusScanState *scanState, Task *task)
 			placementAccessList = list_make1(placementAccess);
 		}
 
-		if (placementAccessList == NIL)
```

Comment:
Not possible, because of preceding checks.

Reply:
The error below suggests a condition under which it is possible (FindShardPlacementOnGroup in BuildPlacementAccessList returns NULL because a shard move completed in between planning and execution). We should remove `Assert(list_length(placementAccessList) == list_length(relationShardList));` instead.